torchmultimodal/architectures/clip.py

```diff
@@ -20,32 +22,38 @@ class CLIPArchitecture(nn.Module):
     encoders, while the loss is implemented in ContrastiveLossWithTemperature.
 
 
-    Args:   vision_encoder (nn.Module): Instantiated vision encoder.
-                See e.g. ResNetForCLIP class.
-            text_encoder (nn.Module): Instantiated text encoder.
-                See CLIPTextEncoder class.
+    Args:   encoders (nn.ModuleDict): Dict of instantiated encoders, keyed by modality.
+                E.g. {"vision": ResNetForCLIP(), "text": CLIPTextEncoder()}
 
-    Inputs: image (Tensor): Tensor containing image features.
-            text (Tensor): Tensor containing text features.
+    Inputs: modalities (Dict[str, Tensor]): Dict of Tensor features, keyed by modality.
+                Must contain one entry for every modality in ``encoders``.
+
+    Output: CLIPOutput object with fields ``{modality}_embeddings`` for every modality
+                in ``encoders``.
     """
 
     def __init__(
         self,
-        vision_encoder: nn.Module,
-        text_encoder: nn.Module,
+        encoders: nn.ModuleDict,
     ):
         super().__init__()
-        self.vision_encoder = vision_encoder
-        self.text_encoder = text_encoder
+        self.encoders = nn.ModuleDict({k: encoders[k] for k in sorted(encoders.keys())})
 
     def forward(
         self,
-        image: torch.Tensor,
-        text: torch.Tensor,
+        modalities: Dict[str, torch.Tensor],
     ) -> Dict[str, torch.Tensor]:
-
-        img_embeddings = self.vision_encoder(image)
-        text_embeddings = self.text_encoder(text)
-        img_embeddings = F.normalize(img_embeddings)
-        text_embeddings = F.normalize(text_embeddings)
-        return {"image": img_embeddings, "text": text_embeddings}
+        embeddings = {}
+        for key, encoder in self.encoders.items():
+            assert key in modalities, f"{key} missing in input"
```

Comment:
i think a ValueError is more fitting since the input is missing a key, rather than the input containing a bad key, so I've changed the assertionerror to a valueerror